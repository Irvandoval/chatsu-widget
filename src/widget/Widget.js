import React, { Component } from 'react'
import Frame from './Frame';
import FloatingButton from './FloatingButton';
import Header from './Header';
import ArrowIcon from './ArrowIcon';
import {
  desktopTitleStyle, 
  desktopWrapperStyle,
  mobileOpenWrapperStyle, 
  mobileClosedWrapperStyle,
  desktopClosedWrapperStyleChat
} from "./styles";

export default class Widget extends Component {

  constructor(props) {
    super(props);
    this.state = {isChatOpen: false};
    this.onClick = this.onClick.bind(this);
  }

  onClick() {
    let stateData = {
      isChatOpen: !this.state.isChatOpen,
    }
    this.setState(stateData);
  };

  render() {
    const conf = this.props.conf;

    const wrapperWidth = {width: conf.desktopWidth};
    const desktopHeight = (window.innerHeight - 100 < conf.desktopHeight) ? window.innerHeight - 90 : conf.desktopHeight;
    // const wrapperHeight = {height: desktopHeight};

    let wrapperStyle;
    if (!this.state.isChatOpen && (this.props.isMobile || conf.alwaysUseFloatingButton)) {
      wrapperStyle = {...mobileClosedWrapperStyle}; // closed mobile floating button
    } else if (!this.props.isMobile) {
      wrapperStyle = (conf.closedStyle === 'chat' || this.state.isChatOpen) ?
        (this.state.isChatOpen) ?
          { ...desktopWrapperStyle, ...wrapperWidth} : // desktop mode, button style
          { ...desktopWrapperStyle}
        :
        { ...desktopClosedWrapperStyleChat}; // desktop mode, chat style
    } else {
      wrapperStyle = mobileOpenWrapperStyle; // open mobile wrapper should have no border
    }

    return (
      <div style={wrapperStyle}>
        { /* Open/close button */ }
        { (this.props.isMobile || conf.alwaysUseFloatingButton) && !this.state.isChatOpen ?

            <FloatingButton color={conf.mainColor} onClick={this.onClick}/>

            :

            (conf.closedStyle === 'chat' || this.state.isChatOpen) ?
              <div style={{background: conf.mainColor, ...desktopTitleStyle}} onClick={this.onClick}>
                <div style={{display: 'flex', alignItems: 'center', padding: '0px 30px 0px 0px'}}>
                  {this.state.isChatOpen ? conf.titleOpen : conf.titleClosed}
                </div>
                <ArrowIcon isOpened={this.state.isChatOpen}/>
              </div>
              :
              <Header onClick={this.onClick} conf={conf}/>
        }

        { /*Chat iFrame */ }
        <div style={{display: this.state.isChatOpen ? 'block' : 'none', height: this.props.isMobile ? '100%' : desktopHeight}}>
          <Frame {...this.props}></Frame>
        </div>

      </div>
    )
  }
};
