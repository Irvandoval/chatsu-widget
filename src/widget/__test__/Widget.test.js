import React from 'react';
import Widget from '../Widget';
import { shallow } from 'enzyme';

const conf ={};

describe('<Widget />', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<Widget conf={conf}/>);
    expect(tree).toMatchSnapshot();
  });
})