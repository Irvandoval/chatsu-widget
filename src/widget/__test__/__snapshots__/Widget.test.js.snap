// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Widget /> matches the snapshot 1`] = `
<div
  style={
    Object {
      "bottom": "0px",
      "boxSizing": "content-box",
      "minHeight": "120px",
      "minWidth": "400px",
      "overflow": "hidden",
      "position": "fixed",
      "right": "0px",
      "zIndex": 2147483647,
    }
  }
>
  <Header
    conf={Object {}}
    onClick={[Function]}
  />
  <div
    style={
      Object {
        "display": "none",
        "height": undefined,
      }
    }
  >
    <Frame
      conf={Object {}}
    />
  </div>
</div>
`;
